darwin/DarwinProcess.c

```diff
@@ -330,30 +344,34 @@ void DarwinProcess_setFromKInfoProc(Process* proc, const struct kinfo_proc* ps,
    proc->updated = true;
 }
 
-void DarwinProcess_setFromLibprocPidinfo(DarwinProcess* proc, DarwinProcessList* dpl, double time_interval) {
+void DarwinProcess_setFromLibprocPidinfo(DarwinProcess* proc, DarwinProcessList* dpl, double time_interval_ns) {
    struct proc_taskinfo pti;
 
    if (sizeof(pti) == proc_pidinfo(proc->super.pid, PROC_PIDTASKINFO, 0, &pti, sizeof(pti))) {
-      uint64_t total_existing_time = proc->stime + proc->utime;
-      uint64_t total_current_time = pti.pti_total_system + pti.pti_total_user;
+      uint64_t total_existing_time_ns = proc->stime + proc->utime;
+
+      uint64_t user_time_ns = machTicksToNanoseconds(pti.pti_total_user);
+      uint64_t system_time_ns = machTicksToNanoseconds(pti.pti_total_system);
```

Comment:
The values of `pti_total_user` and `pti_total_system` that comeback from Libproc (specifically `proc_pidinfo` above) are in mach timebase ticks.